packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/EditableTableCell.tsx

```diff
@@ -37,102 +39,143 @@ export function EditableTableCell({ name, table, keyTuple, value: defaultValue }
-      console.error("Error:", error);
-      toast.error(error.message || "Something went wrong. Please try again.", {
-        id: context?.toastId,
-      });
-      setValue(defaultValue);
+      try {
+        const fieldIndex = getFieldIndex<ValueSchema>(getSchemaTypes(valueSchema), name);
+        const encodedFieldValue = encodeField(fieldType, value);
+        const writeArgs = {
+          abi: IBaseWorldAbi,
+          address: worldAddress as Hex,
+          functionName: "setField",
+          args: [table.tableId, keyTuple, fieldIndex, encodedFieldValue],
+          chainId,
+        } as const;
+
+        await simulateContract(wagmiConfig, {
+          ...writeArgs,
+          account: account.address,
+        });
+
+        const txHash = await writeContract(wagmiConfig, writeArgs);
+        const receipt = await waitForTransactionReceipt(wagmiConfig, { hash: txHash });
+        if (receipt.status !== "success") {
+          throw new Error("Transaction reverted. Please try again.");
+        }
+
+        toast.success(
+          <a href={blockExplorerTransactionUrl({ hash: txHash, chainId })} target="_blank" rel="noopener noreferrer">
+            Transaction successful: {txHash} <ExternalLinkIcon className="inline-block h-3 w-3" />
+          </a>,
+        );
+        queryClient.invalidateQueries({
+          queryKey: [
+            "balance",
+            {
+              address: account.address,
+              chainId,
+            },
+          ],
+        });
+
+        return { txHash, receipt };
+      } catch (error) {
+        console.error("Error:", error);
+        toast.error(
+          error instanceof Error ? error.message : String(error) || "Something went wrong. Please try again.",
+        );
+        throw error;
+      }
     },
   });
 
-  const handleSubmit = (newValue: unknown) => {
-    if (!account.isConnected) {
-      return openConnectModal?.();
+  // When the indexer has picked up the successful write, we can clear the write result
+  useEffect(() => {
+    if (write.status === "success" && BigInt(blockHeight) >= write.data.receipt.blockNumber) {
+      write.reset();
     }
-    mutate(newValue);
-  };
-
-  if (fieldType === "bool") {
-    return (
-      <div className="flex items-center gap-1">
-        <Checkbox
-          id={`checkbox-${name}`}
-          checked={Boolean(value)}
-          onCheckedChange={handleSubmit}
-          disabled={isPending}
-        />
-        {isPending && <LoaderIcon className="h-4 w-4 animate-spin" />}
-      </div>
-    );
-  }
+  }, [write, blockHeight]);
+
+  const [edit, setEdit] = useState<{
+    value: string;
+    initialValue: string;
+  } | null>(null);
 
   return (
     <div className="w-full">
-      {!isPending && (
-        <form
-          onSubmit={(evt) => {
-            evt.preventDefault();
-            handleSubmit(value);
-          }}
-        >
+      <form
+        ref={formRef}
+        onSubmit={(event) => {
```

Comment:
I know I said in my PR that an improvement would be to wrap this entire thing in a form but I feel like this has made it harder to reason about because we're not using the form as a form but rather just a way to trigger the mutation

what I meant was that we could use the form like a real HTML form with field values, then use those values in the submit handler to perform the write (like a real form would), then we get all the benefits of the DOM like accessibility, browser APIs, etc.